packages/server/src/cloud/aws/deploy.ts

```diff
@@ -141,18 +180,22 @@ async function lambdaExists(client: LambdaClient, name: string): Promise<boolean
     const command = new GetFunctionCommand({ FunctionName: name });
     const response = await client.send(command);
     return response.Configuration?.FunctionName === name;
-  } catch (_err) {
-    return false;
+  } catch (err) {
+    if (err instanceof ResourceNotFoundException) {
```

Comment:
Before we swallowed all errors and returned false, we now catch `ResourceNotFoundException` which is returned when the Lambda doesn't already exist, and rethrow all other errors